apps/events/api/views.py

```diff
@@ -0,0 +1,295 @@
+        return Response(data=serializer.data, status=status.HTTP_200_OK)
+
+    @action(
+        detail=True,
+        methods=["GET"],
+        permission_classes=(permissions.IsAuthenticated,),
+        serializer_class=ExtrasSerializer,
+    )
+    def extras(self, request, pk=None):
+        attendance_event: AttendanceEvent = self.get_object()
+        serializer = self.get_serializer(attendance_event.extras, many=True)
+        return Response(data=serializer.data, status=status.HTTP_200_OK)
+
+    @action(
+        detail=True,
+        methods=["GET"],
+        permission_classes=(permissions.IsAuthenticated,),
+        serializer_class=PaymentReadOnlySerializer,
+    )
+    def payment(self, request, pk=None):
+        attendance_event: AttendanceEvent = self.get_object()
+        payment = attendance_event.get_payment()
+        if not payment:
+            raise NotFound
+        serializer = self.get_serializer(payment)
+        return Response(data=serializer.data, status=status.HTTP_200_OK)
+
+
+class AttendeeViewSet(
+    viewsets.GenericViewSet, mixins.ListModelMixin, mixins.RetrieveModelMixin
+):
+    serializer_class = AttendeeSerializer
+    filterset_fields = (
+        "event",
+        "attended",
+        "user",
+        "show_as_attending_event",
+        "allow_pictures",
+        "extras",
+    )
+
+    @staticmethod
+    def _get_allowed_attendees(user):
+        """
+        A user is allowed to see attendees for their own user, and for events they are organizing.
+        """
+        attendees = get_objects_for_user(
+            user, "events.change_attendee", accept_global_perms=False
+        )
+        attendees |= Attendee.objects.filter(user=user)
+        return attendees.distinct()
+
+    def get_queryset(self):
+        return self._get_allowed_attendees(self.request.user)
+
+    @action(
+        detail=True,
+        methods=["PATCH", "PUT"],
+        permission_classes=(ChangeAttendeePermission,),
+        serializer_class=AttendeeUpdateSerializer,
+    )
+    def change(self, request, pk=None):
+        attendee: Attendee = self.get_object()
+        partial = request.method == "PATCH"
+        serializer = self.get_serializer(attendee, data=request.data, partial=partial)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(data=serializer.data, status=status.HTTP_200_OK)
+
+    @action(
+        detail=True,
+        methods=["PATCH", "PUT"],
+        permission_classes=(permissions.DjangoObjectPermissions,),
+        serializer_class=AttendeeAdministrateSerializer,
+    )
+    def administrate(self, request, pk=None):
+        attendee: Attendee = self.get_object()
+        partial = request.method == "PATCH"
+        serializer = self.get_serializer(attendee, data=request.data, partial=partial)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(data=serializer.data, status=status.HTTP_200_OK)
```

Comment:
So you have to be an event admin to be able to call this endpoint at all, that is what the permission does. The serializer for this endpoint lets you edit a specific set of fields that admins can edit.
This endpoint would be called from something like dashboard, where you are in the role of an admin. This means there are specific things you can't do from this endpoint that you are actually allowed to do since it is your user. You would have to call another endpoint, just like any other user to change a consent for example.